docs/tutorials/entrypoints.rst

```diff
@@ -8,10 +8,96 @@
 Entry points
 ************
 
-
 .. todo::
 
-   - Mention what is the purpose of entry points
-   - Give an example of a console entry point
    - Give an example of a custom entry point
      - Mention pluggy maybe for an example use-case?
+
+Introduction
+============
+
+Entry points provide a mechanism to advertise components of an installed
+distribution to other code or users. Most notably, the ``console_scripts``
```

Comment:
I'd avoid using `console_scripts`, because it's what `setuptools` uses and using that in `pyproject.toml` is an error (the table is called `project.scripts`. So I'd say something like "there are three type of entry-points: those that provide commands to execute in a shell, commands to launch a GUI, and discoverable (plugin style) entry-points.